src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ObjectCollection.cs

```diff
@@ -388,9 +388,17 @@ public void RemoveAt(int index)
                 OwnerComboBoxAccessibleObject?.ItemAccessibleObjects.Remove(InnerList[index]);
                 InnerList.RemoveAt(index);
 
-                if (!_owner.IsHandleCreated && index < _owner._selectedIndex)
+                if (!_owner.IsHandleCreated)
                 {
-                    _owner._selectedIndex--;
+                    if (index < _owner._selectedIndex)
+                    {
+                        _owner._selectedIndex--;
+                    }
+                    else if (index == _owner._selectedIndex)
+                    {
+                        _owner._selectedIndex = -1;
+                        _owner.UpdateText();
```

Comment:
Probably events should also be triggered:
https://github.com/dotnet/winforms/blob/e83b710bb8bbf9c9c40395e2d070ff7d5f3e80b4/src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.cs#L1035-L1036
```suggestion
                        // Update ComboBox text and trigger selection changed events
                        _owner.SelectedIndex = -1;
```